src/test/utils/s3ops.js

```diff
@@ -673,15 +675,22 @@ class S3OPS {
         }
     }
 
-    async get_object(bucket, key) {
+    async get_object(bucket, key, query_params) {
         const params = {
             Bucket: bucket,
             Key: key
         };
         console.log('Reading object ', key);
         try {
             //There can be an issue if the object size (length) is too large
-            const obj = await this.s3.getObject(params).promise();
+            const obj = await this.s3.getObject(params)
+            .on('build', req => {
+                if (query_params) {
+                    const query_string = querystring.stringify(query_params);
+                    console.log(`Setting query parameter ${query_string}`);
+                    req.httpRequest.path += `?${query_string}`;
+                }
```

Comment:
I think you still need to use the same condition as you had before in case the path already contains a search string, something like:
```suggestion
                if (!query_params) return;
                const sep = req.httpRequest.search() ? '&' : '?';
                const query_string = querystring.stringify(query_params);
                const req_path = `${req.httpRequest.path}${sep}${query_string}`;
                console.log(`Added query params ${query_params} to path ${req_path}`);
                req.httpRequest.path = req_path;
```

Reply:
My bad. Should have checked whether request has existing query parameter like before.